### PR TITLE
PVC integration tests: TestBackup and TestRegularWorkspaceTasks and TestWorkspaceEnableAndDisablePVC

### PR DESCRIPTION
**Pull request by jenting**
## Description

Add PVC feature flag to TestBackup
- Create a fresh new workspace with Git initializer
- Write a `foobar.txt` file
- Stop the workspace
- Recreate the workspace
- Make sure the `foobar.txt` file exists

Add PVC feature flag to TestRegularWorkspaceTasks
- Create a fresh new workspace with Git initializer and Gitpod tasks
- Make sure each task generated file exists

Add PVC integration test to TestWorkspaceEnableAndDisablePVC
- Open a fresh new workspace
- Write a `foobar.txt` file
- Stop the workspace
- Relaunch the workspace with the PVC feature flag
- Make sure the `foobar.txt` file exists
- Stop the workspace
- Relaunch the workspace without the PVC feature flag
- Make sure the `foobar.txt` file exists

## Related Issue(s)

#12497

## How to test

```console
cd test

go test  -run ^TestBackup$ github.com/gitpod-io/gitpod/test/tests/components/ws-manager -count=1 -namespace=default -kubeconfig=/home/gitpod/.kube/config -v

go test  -run ^TestRegularWorkspaceTasks$ github.com/gitpod-io/gitpod/test/tests/components/ws-manager -count=1 -namespace=default -kubeconfig=/home/gitpod/.kube/config -v

go test  -run ^TestWorkspaceEnableAndDisablePVC$ github.com/gitpod-io/gitpod/test/tests/components/ws-manager -count=1 -namespace=default -kubeconfig=/home/gitpod/.kube/config -v
```

## Release Notes

```release-note
None
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: test/pkg/integration/integration.go [hunk 1]
```diff
@@ -501,3 +503,27 @@ func isPodReady(s *corev1.PodStatus) bool {
...
 	return false
 }
+
+func FindVolumeSnapshot(instanceID, namespace string, client klient.Client) (string, string, error) {
+	vsClient, err := volumesnapshotv1.NewForConfig(client.RESTConfig())
+	if err != nil {
+		return "", "", err
+	}
+
+	volumeSnapshots, err := vsClient.SnapshotV1().VolumeSnapshots(namespace).List(context.Background(), metav1.ListOptions{
```
Comment (utam0k) [inline test/pkg/integration/integration.go:513]: Is this client a sure success? Because pod port forwarding etc. is unstable and will try to retry.

Comment (jenting) [reply test/pkg/integration/integration.go:513]: It is not using the port-forward function https://github.com/gitpod-io/gitpod/blob/2871bdc1b853dbdef502c54cf950e7319babef61/test/pkg/integration/common/port_forward.go#L31

It's a client that uses kubeconfig to access the Kubernetes cluster.

Comment (utam0k) [reply test/pkg/integration/integration.go:513]: The question was not good. Is this client stable in the face of unstable networks and lack of resources?

Comment (jenting) [reply test/pkg/integration/integration.go:513]: Ugh, if the network is unstable _or_ the kube-apiserver is out of resources, then yes, the client can't connect to the server.

Comment (utam0k) [reply test/pkg/integration/integration.go:513]: I'm not saying perfect, but what about adding retrying to tolerate such an environment?

Comment (jenting) [reply test/pkg/integration/integration.go:513]: I see. Currently, all the kubernetes client does not implement the retry within the integration tests https://github.com/gitpod-io/gitpod/blob/main/test/pkg/integration/integration.go, can we handle it in other PR?

Comment (utam0k) [reply test/pkg/integration/integration.go:513]: Make sense

**Comment by utam0k** [comment]
Waiting for https://werft.gitpod-dev.com/job/gitpod-custom-jenting-12497.0

File: test/pkg/integration/integration.go (hunk 1 above)
Comment (utam0k) [inline test/pkg/integration/integration.go:513]: How about getting `ctx` from the caller of `FindVolumeSnaphot`?

**Comment by utam0k** [comment]
@jenting I made a commend, but other things look good to me.

File: test/pkg/integration/integration.go (hunk 1 above)
Comment (jenting) [reply test/pkg/integration/integration.go]: Nice!

**Review by utam0k — APPROVED** [review]
Thanks!

**Pull request merged**